Branch-TestBed/Branch-TestBed/AppDelegate.m

```diff
@@ -52,10 +62,45 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
             NSLog(@"Branch TestBed: Initialization failed\n%@", error.localizedDescription);
         }
     }];
-    
+
     return YES;
 }
 
+- (void)onboardUserOnInstall {
+    NSURL *urlForOnboarding = [NSURL URLWithString:@"http://example.com"]; // Put your onboarding link here
+    
+    id notInstall = [[NSUserDefaults standardUserDefaults] objectForKey:@"notInstall"];
+    if (!notInstall) {
+        [[NSUserDefaults standardUserDefaults] setObject:@YES forKey:@"notInstall"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+
+        Branch *branch = [Branch getInstance];
+
+        // Note that this must be invoked *before* initSession
+        [branch enableDelayedInit];
+
+        NSURL *updatedUrlForOnboarding = [branch getUrlForOnboardingWithRedirectUrl:urlForOnboarding.absoluteString];
+        if (updatedUrlForOnboarding) {
+            // replace url for onboarding with the URL provided by Branch
+            urlForOnboarding = updatedUrlForOnboarding;
+        }
+        else {
+            // do not replace url for onboarding
+            NSLog(@"Was unable to get onboarding URL from Branch SDK, so proceeding with normal onboarding URL.");
+            [branch disableDelayedInit];
+        }
+
+        self.onboardingVC = [[SFSafariViewController alloc] initWithURL:urlForOnboarding];
+        self.onboardingVC.delegate = self;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [[[[UIApplication sharedApplication].delegate window] rootViewController] presentViewController:self.onboardingVC animated:YES completion:NULL];
+        });
+    }
+}
+
+- (void)safariViewControllerDidFinish:(SFSafariViewController *)controller {
```

Comment:
How about we implement the example to trigger init after the `updatedUrlForOnboarding` is finished loading and it's redirecting to the original URL?